### PR TITLE
middleware for validating nostr event payload 

### PR DESCRIPTION
**Pull request by mubarak23**

### CODE REVIEW

File: cmd/server/main.go [hunk 1]
```diff
@@ -154,11 +154,19 @@ func main() {
 	// strict rate limit for requests for sending payments
 	strictRateLimitMiddleware := transport.CreateRateLimitMiddleware(c.StrictRateLimit, c.BurstRateLimit)
 	secured := e.Group("", tokens.Middleware(c.JWTSecret), logMw)
+
+	// Appying the custom middleware to a Group
+	validateNostrPayload := e.Group("", svc.ValidateEventPayload(), logMw)
```
Comment (Nsandomeno) [inline cmd/server/main.go:159]: @mubarak23 one thing to note, is that this group may also want to (optionally) apply the tokens.Middleware as seen applied to the Group assigned to the secured variable.

When we see the content CREATE_USER we do not need to ensure a valid JWT, however for any other event we will want to ensure they have the JWT on-hand... Please let me know if that makes sense.

File: cmd/server/main.go [hunk 2]
```diff
@@ -154,11 +154,19 @@ func main() {
...
+
 	securedWithStrictRateLimit := e.Group("", tokens.Middleware(c.JWTSecret), strictRateLimitMiddleware, logMw)
 
 	transport.RegisterLegacyEndpoints(svc, e, secured, securedWithStrictRateLimit, strictRateLimitMiddleware, tokens.AdminTokenMiddleware(c.AdminToken), logMw)
 	transport.RegisterV2Endpoints(svc, e, secured, securedWithStrictRateLimit, strictRateLimitMiddleware, tokens.AdminTokenMiddleware(c.AdminToken), logMw)
 
+	// add it the transport layer
+	transport.RegisterNoStrevent(svc, e, validateNostrPayload)
```
Comment (Nsandomeno) [inline cmd/server/main.go:167]: Is there a path to adding the new endpoint in with the other groups passed to RegisterV2Endpoints, by adding a new parameter for the group you created i.e. `validateNostrPayload`?

File: controllers_v2/notstr.ctrl.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+func NewNoStrController(svc *service.LndhubService) *NoStrController {
+	return &NoStrController{svc: svc}
+}
+
+
+type AddNoStrResponseBody struct  {
+	ID        string `json:"ID"`
+	Pubkey    string `json:"Pubkey"`
+	Kind      int    `json:"kind"`
+	Ta        string `json:"Ta"`
```
Comment (Nsandomeno) [inline controllers_v2/notstr.ctrl.go:25]: Our NIP-1 custom tags (ta, amt, fee, addr etc. ) should be nullable. However, no other field should be nullable

File: lib/service/service.go [hunk 4]
```diff
@@ -88,3 +90,109 @@ func (svc *LndhubService) ParseInt(value interface{}) (int64, error) {
 		return 0, fmt.Errorf("conversion to int from %T not supported", v)
 	}
 }
+
+
+func (svc *LndhubService) ValidateEventPayload () echo.MiddlewareFunc {
```
Comment (Nsandomeno) [inline lib/service/service.go:95]: Great work to prepare us for next steps here!

File: cmd/server/main.go (hunk 1 above)
Comment (mubarak23) [reply cmd/server/main.go:159]: Sure, this make sense, we can pass the token in the group, then add the token middleware to the specific endpoint

File: lib/transport/nostr_event.go [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package transport
+
+import (
+	v2controllers "github.com/getAlby/lndhub.go/controllers_v2"
+	"github.com/getAlby/lndhub.go/lib/service"
+	"github.com/labstack/echo/v4"
+)
+
+func RegisterNoStrevent(svc *service.LndhubService, e *echo.Echo, validateNostr *echo.Group) {
```
Comment (Nsandomeno) [inline lib/transport/nostr_event.go:9]: think we mentioned we could delete this after moving to v2_endpoints transport

Comment (Nsandomeno) [reply lib/transport/nostr_event.go:9]: `git rm -rf lib/transport/nostr_event.go && git add . && git commit -m "del" && git push`

File: cmd/server/main.go [hunk 6]
```diff
@@ -153,11 +153,19 @@ func main() {
 	logMw := transport.CreateLoggingMiddleware(logger)
 	// strict rate limit for requests for sending payments
 	strictRateLimitMiddleware := transport.CreateRateLimitMiddleware(c.StrictRateLimit, c.BurstRateLimit)
-	secured := e.Group("", tokens.Middleware(c.JWTSecret), svc.ValidateUserMiddleware(), logMw)
-	securedWithStrictRateLimit := e.Group("", tokens.Middleware(c.JWTSecret), svc.ValidateUserMiddleware(), strictRateLimitMiddleware, logMw)
+
+	secured := e.Group("", tokens.Middleware(c.JWTSecret), logMw)
+
+	// Appying the custom middleware to a Group
+	validateNostrPayload := e.Group("", svc.ValidateNosTREventPayload(), logMw)
```
Comment (Nsandomeno) [inline cmd/server/main.go:160]: Consistent casing / naming : can we please change `ValidateNosTREventPayload` to `ValidateNostrEventPayload` for consistency?

File: controllers_v2/notstr.ctrl.go [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+import (
+	"net/http"
+
+	"github.com/getAlby/lndhub.go/lib/responses"
+	"github.com/getAlby/lndhub.go/lib/service"
+	"github.com/labstack/echo/v4"
+)
+
+// NoStrController : Add NoStr Event controller struct
+type NoStrController struct {
```
Comment (Nsandomeno) [inline controllers_v2/notstr.ctrl.go:12]: Consistent casing / naming : can we please change `NoStrController` to `NostrController` for consistency?

**Pull request merged**